doc/tutorials/using_partition_mode_on_up2.rst

```diff
@@ -2,154 +2,312 @@
+   the ACRN serial console to switch between privileged VMs.
 
-     cd ~/acrn-hypervisor/hypervisor
-     make menuconfig
+   .. code-block:: none
+     :emphasize-lines: 1
 
-  Select entry ``Hypervisor mode (Sharing mode)`` and press ``Enter``.
-  Then select ``Partition mode`` sub-entry.
+     $ dmesg | grep ttyS
+     [   10.552369] dw-apb-uart.8: ttyS4 at MMIO 0x91524000 (irq = 4, base_baud = 115200) is a 16550A
+     [   10.552568] serial serial0: tty port ttyS4 registered
+     [   10.572344] dw-apb-uart.9: ttyS5 at MMIO 0x91522000 (irq = 5, base_baud = 115200) is a 16550A
+     [   10.572545] serial serial1: tty port ttyS5 registered
 
-* Configure the ACRN hypervisor's ``serial base address``
+   The following command prints detailed information about all PCI buses
+   and devices in the system. Look up the PCI BAR addresses of the SATA
+   controller and the USB controller, record down these addresses for
+   editing the source code later.
 
-  .. code-block:: none
+   .. code-block:: none
+     :emphasize-lines: 1,9,10,14,22
+
+     $ sudo lspci -vv
+     ...
+     00:12.0 SATA controller: Intel Corporation Device 5ae3 (rev 0b) (prog-if 01 [AHCI 1.0])
+         Subsystem: Intel Corporation Device 7270
+         Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
+         Status: Cap+ 66MHz+ UDF- FastB2B+ ParErr- DEVSEL=medium >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
+         Latency: 0
+         Interrupt: pin A routed to IRQ 123
+         Region 0: Memory at 91514000 (32-bit, non-prefetchable) [size=8K]
+         Region 1: Memory at 91537000 (32-bit, non-prefetchable) [size=256]
+         Region 2: I/O ports at f090 [size=8]
+         Region 3: I/O ports at f080 [size=4]
+         Region 4: I/O ports at f060 [size=32]
+         Region 5: Memory at 91536000 (32-bit, non-prefetchable) [size=2K]
+     ...
+     00:15.0 USB controller: Intel Corporation Device 5aa8 (rev 0b) (prog-if 30 [XHCI])
+         Subsystem: Intel Corporation Device 7270
+         Control: I/O- Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
+         Status: Cap+ 66MHz- UDF- FastB2B+ ParErr- DEVSEL=medium >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
+         Latency: 0
+         Interrupt: pin A routed to IRQ 122
+         Region 0: Memory at 91500000 (64-bit, non-prefetchable) [size=64K]
+
+#. Clone the ACRN source code [1]_ and configure the build options with
+   ``make menuconfig`` command:
 
-     cd ~/acrn-hypervisor/hypervisor
-     make menuconfig
+   .. code-block:: none
+     :emphasize-lines: 1,3
 
-  Select entry ``Base address of serial MMIO region`` and press ``Enter``.
-  Then we will see a prompt msgbox with title ``Value for `Base address of serial MMIO region` (hex)``.
-  Please type the serial base mmio address of your UP2 board and press ``Enter``.
+     $ git clone https://github.com/projectacrn/acrn-hypervisor.git
```

Comment:
This clone command gets you the master branch, but your (very subtle) footnote says to use a particular commit ID. If you intend to have folks NOT use the master branch's code, put that information here and in a more visible way.  What commands should a user use to get the commit ID code you've tested?

Reply:
@dbkinder I'd suggest the user always use the code on master branch, rollback to our *know-to-work* commits only when there hit regressions. In case they want to try our tested commits, they can 'git checkout <commit>' after the clone command. How do you want to proceed?